ciphers/elliptic_curve_key_exchange.cpp

```diff
@@ -0,0 +1,302 @@
+/**
+ * @file elliptic_curve_key_exchange.cpp
+ * @brief Implementation of [Elliptic Curve Diffie Hellman Key
+ * Exchange](https://cryptobook.nakov.com/asymmetric-key-ciphers/ecdh-key-exchange).
+ *
+ * @details
+ * The ECDH (Elliptic Curve Diffie–Hellman Key Exchange) is anonymous key
+ * agreement scheme, which allows two parties, each having an elliptic-curve
+ * public–private key pair, to establish a shared secret over an insecure
+ * channel.
+ * ECDH is very similar to the classical DHKE (Diffie–Hellman Key Exchange)
+ * algorithm, but it uses ECC point multiplication instead of modular
+ * exponentiations. ECDH is based on the following property of EC points:
+ * (a * G) * b = (b * G) * a
+ * If we have two secret numbers a and b (two private keys, belonging to Alice
+ * and Bob) and an ECC elliptic curve with generator point G, we can exchange
+ * over an insecure channel the values (a * G) and (b * G) (the public keys of
+ * Alice and Bob) and then we can derive a shared secret:
+ * secret = (a * G) * b = (b * G) * a.
+ * Pretty simple. The above equation takes the following form:
+ * alicePubKey * bobPrivKey = bobPubKey * alicePrivKey = secret
+ * @author [Ashish Daulatabad](https://github.com/AshishYUO)
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <iomanip>
+#include <iostream>
+
+#include "uint256_t.hpp"
```

Comment:
Add a one-line description of what the library/header is for (see the example below).
```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```